org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/SocketServerLauncher.xtend

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2019 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+ 
+package org.eclipse.xtext.ide.server
+
+import com.google.inject.Guice
+import java.io.PrintWriter
+import java.net.InetSocketAddress
+import java.nio.channels.AsynchronousServerSocketChannel
+import java.nio.channels.Channels
+import org.eclipse.lsp4j.jsonrpc.Launcher
+import org.eclipse.lsp4j.services.LanguageClient
+import org.eclipse.xtext.ide.server.LanguageServerImpl
+import org.eclipse.xtext.ide.server.ServerModule
+import org.eclipse.xtext.util.internal.Log
+
+/**
+ * Allows to launch a language server vie a server socket.
+ * 
+ * Useful for debugging live language servers.
+ * 
+ * @author Jan K&ouml;hnlein - Initial contribution and API
+ * @since 2.18
+ */
+@Log
+class SocketServerLauncher {
+
+	static val HOST = '-host'
+	static val PORT = '-port'
+	static val TRACE = '-trace'
+	static val NO_VALIDATE = '-noValidate'
```

Comment:
should there be a log option analogous to server laucnher

Reply:
Logging is always enabled, as this is a debug scenario. In the server launcher the `log` option mainly serves for redirecting the logs to not pollute`stdin` and `stdout` which are used for JSON-RPC. I don't think these have to be 100% symmetric.